pkg/aws/ec2/ec2.go

```diff
@@ -195,8 +195,8 @@ func parseENI(iface *ec2.NetworkInterface, vpcs ipamTypes.VirtualNetworkMap, sub
 	}
 
 	for _, ip := range iface.PrivateIpAddresses {
-		if ip.PrivateIpAddress != nil {
-			eni.Addresses = append(eni.Addresses, *ip.PrivateIpAddress)
+		if ip.PrivateIpAddress != nil && !ip.Primary {
+			eni.Addresses = append(eni.Addresses, aws.ToString(ip.PrivateIpAddress))
```

Comment:
This did not change in the upstream patch. Please revert the change to this line.
```suggestion
			eni.Addresses = append(eni.Addresses, *ip.PrivateIpAddress)
```